comp/api/api/def/component.go

```diff
@@ -22,9 +22,8 @@ import (
 
 // Component is the component type.
 type Component interface {
-	StartServer() error
-	StopServer()
-	ServerAddress() *net.TCPAddr
+	CMDServerAddress() *net.TCPAddr
+	IPCServerAddress() *net.TCPAddr
```

Comment:
It seems that we are adding these two new functions that are used in the test and `pkg/cli/standalone/jmx.go`.

Wondering if we could avoid that and get the values from the config component. It might require a bit more refactoring, but we avoid the addition of function on exiting interfaces when we can use the config component which is ubiquitous in most of the places 😄

Reply:
In some case the port used by the API is not equal to the one set in the configuration (for example if we set `cmd_host` to `0` the OS will pick one free.
So there is two solutions for this:
- avoid to use value like `0` -> enforce consistance between config and API real port
- allow `0` values ->  it prevents to use it for example in unit tests, where all we need is a free port.

What do you think ?

Reply:
We discussed IRL and we decided to leave the code the way it is for now and on work on removing those function in a follow up PR if some of our assumptions around JMX are correct 😄